.github/workflows/build.yml

```diff
@@ -0,0 +1,143 @@
+
+    - name: Install Terraform CloudFoundry Provider
+      run: |
+          mkdir -p $HOME/.terraform.d/plugins/linux_amd64
+          wget -O ${{ env.CF_PROVIDER }} https://github.com/cloudfoundry-community/terraform-provider-cf/releases/latest/download/terraform-provider-cloudfoundry_linux_amd64
+          chmod +x ${{ env.CF_PROVIDER }}
+      env:
+        CF_PROVIDER: $HOME/.terraform.d/plugins/linux_amd64/terraform-provider-cloudfoundry
+
+    - name: Terraform init
+      run: |
+          terraform init \
+           -backend-config=storage_account_name=${{ env.BACKEND_STORAGE_ACCOUNT_NAME }} \
+           -backend-config=access_key=${{ env.BACKEND_ACCESS_KEY }} \
+           -backend-config=key=${{ env.BACKEND_KEY }} \
+            ${{ env.PAAS_TF_DIR }}
+      env:
+        BACKEND_STORAGE_ACCOUNT_NAME: s121d01tfstatestr
+        BACKEND_ACCESS_KEY: ${{ secrets.BACKEND_ACCESS_KEY }}
+        BACKEND_KEY: paas-ttapi.tfstate
+        PAAS_TF_DIR: $HOME/work/teacher-training-api/teacher-training-api/terraform/paas
+
+    - name: Terraform apply
+      run: |
+          terraform apply -auto-approve \
+            -var-file=${{ env.PAAS_TF_DIR }}/${{ env.TERRAFORM_VAR_FILE }} \
+            ${{ env.PAAS_TF_DIR }}
+      env:
+        BACKEND_STORAGE_ACCOUNT_NAME: s121d01tfstatestr
+        BACKEND_ACCESS_KEY: ${{ secrets.BACKEND_ACCESS_KEY }}
+        BACKEND_KEY: paas-ttapi.tfstate
+        PAAS_TF_DIR: $HOME/work/teacher-training-api/teacher-training-api/terraform/paas
+        TERRAFORM_VAR_FILE: terraform_qa.tfvars
+        TF_VAR_ttapi_image_tag: ${{env.SHA_TAG}}
+        TF_VAR_api_url: https://api.london.cloud.service.gov.uk
+        TF_VAR_docker_image: ${{ env.DOCKER_IMAGE }}:${{ env.SHA_TAG }}
+        TF_VAR_user: ${{ secrets.TF_VAR_user }}
+        TF_VAR_password: ${{ secrets.TF_VAR_password }}
+        TF_VAR_SECRET_KEY_BASE: ${{ secrets.TF_VAR_SECRET_KEY_BASE }}
+        TF_VAR_SENTRY_DSN: ${{ secrets.TF_VAR_SENTRY_DSN }}
+        # Let's confirm if these variables are required
+        #TF_VAR_SETTINGS__GOOGLE__GCP_API_KEY: ${{ secrets.TF_VAR_SETTINGS__GOOGLE__GCP_API_KEY }}
+        #TF_VAR_SETTINGS__GOOGLE__MAPS_API_KEY: ${{ secrets.TF_VAR_SETTINGS__GOOGLE__MAPS_API_KEY }}
+        #TF_VAR_AUTHENTICATION_TOKEN: ${{ secrets.TF_VAR_AUTHENTICATION_TOKEN }}
+        #TF_VAR_SETTINGS__AUTHENTICATION__SECRET: ${{ secrets.SETTINGS__AUTHENTICATION__SECRET }}
+        #TF_VAR_SETTINGS__GOVUK_NOTIFY__API_KEY: ${{ secrets.SETTINGS__GOVUK_NOTIFY__API_KEY }}
+        #TF_VAR_SETTINGS__GOVUK_NOTIFY__COURSE_PUBLISH_EMAIL_TEMPLATE_ID: ${{ secrets.SETTINGS__GOVUK_NOTIFY__COURSE_PUBLISH_EMAIL_TEMPLATE_ID }}
+        #TF_VAR_SETTINGS__SETTINGS__GOVUK_NOTIFY__COURSE_UPDATE_EMAIL_TEMPLATE_ID: ${{ secrets.SETTINGS__GOVUK_NOTIFY__COURSE_UPDATE_EMAIL_TEMPLATE_ID }}
+        #TF_VAR_SETTINGS__GOVUK_NOTIFY__WELCOME_EMAIL_TEMPLATE_ID: ${{ secrets.SETTINGS__GOVUK_NOTIFY__WELCOME_EMAIL_TEMPLATE_ID }}
```

Comment:
GOVUK Notify variables that are still required:

```ruby
Settings.govuk_notify.api_key
Settings.govuk_notify.course_publish_email_template_id
Settings.govuk_notify.course_sites_update_email_template_id
Settings.govuk_notify.course_subjects_updated_email_template_id
Settings.govuk_notify.course_update_email_template_id
Settings.govuk_notify.course_vacancies_updated_email_template_id
Settings.govuk_notify.course_withdraw_email_template_id
Settings.govuk_notify.magic_link_email_template_id
Settings.govuk_notify.welcome_email_template_id
```

We're working towards pulling these out of the code, but that's a separate exercise. I pulled this list from the code base in master:

```sh
ag Settings.govuk_notify app config | ruby -pe '$_.sub!(/.*(Settings.govuk_notify[.a-z0-9_-]+).*/, "\\1")' | sort -u
```

Reply:
OK let's add them. I assume only `api_key` is a secret?

Reply:
@saliceti Yes `api_key` is a secret and should not be available. The others are merely identifiers and can be public knowledge. In the future we plan to move these identifiers out of the environment variable and just hard code them in the app.

Reply:
Having looked on the docker's environment on Azure and the deploy.json file. The following two keys cant be found:-

* Settings.govuk_notify.course_sites_update_email_template_id
* Settings.govuk_notify.course_subjects_updated_email_template_id

In light of this, are these two keys being used at all?

Reply:
It looks like they have been hard coded here https://github.com/DFE-Digital/teacher-training-api/blob/master/config/settings.yml#L9 which is what we want to move to in the long term. So if they are deleted in azure this should not be a problem